### PR TITLE
utils: add wrapper for the loading cache

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Follow up for #9628
Creates a utility class LazyCache which currently wraps Caffeine library Cache class.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by shwstppr** [comment]
@blueorangutan package

File: utils/src/main/java/org/apache/cloudstack/utils/cache/LazyCache.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+package org.apache.cloudstack.utils.cache;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+
+public class LazyCache<K, V> {
```
Comment (rohityadavcloud) [inline utils/src/main/java/org/apache/cloudstack/utils/cache/LazyCache.java:26]: minor nit - CacheUtils ?

**Review by rohityadavcloud — APPROVED** [review]

File: utils/src/main/java/org/apache/cloudstack/utils/cache/LazyCache.java (hunk 1 above)
Comment (shwstppr) [reply utils/src/main/java/org/apache/cloudstack/utils/cache/LazyCache.java:26]: intentionally didn't use *Utils as we are/will be creating objects of this class unlike other static Utils classes. Any other suggestions maybe?

Comment (DaanHoogland) [reply utils/src/main/java/org/apache/cloudstack/utils/cache/LazyCache.java:26]: How about `org.apache.cloudstack.utils.Cache`, @shwstppr ?

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by rohityadavcloud** [comment]
@blueorangutan test

**Comment by rohityadavcloud** [comment]
@DaanHoogland @JoaoJandre do we need manual testing or is the smoketests good enough? Looks like just refactoring to me.

@blueorangutan test

**Review by sureshanaparti — APPROVED** [review]

**Comment by JoaoJandre** [comment]
@shwstppr @rohityadavcloud why is this needed for 4.20? It seems like it could go in 4.20.1

**Comment by shwstppr** [comment]
@JoaoJandre It is just refactoring and good to have. I'm okay if you want to move it to 4.20.1.

**Comment by shwstppr** [comment]
Closing this as it is part of #9840

**Pull request closed without merging**